README.md

```diff
@@ -1,9 +1,40 @@
 # Conduit Connector Google Cloud Platform Pub/Sub
 
 ### General
-The GCP Pub/Sub connector is one of [Conduit](https://github.com/ConduitIO/conduit) plugins. It provides a source and a destination GCP Pub/Sub connector.
+The GCP Pub/Sub connector is one of [Conduit](https://github.com/ConduitIO/conduit) plugins. It provides both, a source and destination GCP Pub/Sub connector.
 
 ### Prerequisites
 - [Go](https://go.dev/) 1.18
 - (optional) [golangci-lint](https://github.com/golangci/golangci-lint) 1.45.2
-- (optional) [mock](https://github.com/golang/mock) 1.6.0
\ No newline at end of file
+
+### How it works
+
+Under the hood, the connector uses [Google Cloud Client Libraries for Go](https://github.com/googleapis/google-cloud-go).
+
+### Configuration
+The user can get general (for both types of connector) configuration fields from a JSON file by the following instructions: [Getting started with authentication](https://cloud.google.com/docs/authentication/getting-started).
+
+All general fields and fields for a specific connector type are required.
+
+| name             | description                        | type    | example                                                                        |
+|------------------|------------------------------------|---------|--------------------------------------------------------------------------------|
+| `privateKey`     | from JSON, key: `private_key`      | general | -----BEGIN PRIVATE KEY-----\nMIIEvAIBADANBgkqhkiG\n-----END PRIVATE KEY-----\n |
```

Comment:
from json looks strange, please add more detail about it is form json file or some other thing

Reply:
I think I can remove the phrase "from JSON", but to be honest there is nothing to add as a description to these fields.
What do you think?

Reply:
lets add short descriptions, for example
privateKey  : gcp private key for auth to system.
..................